lib/IDE/CodeCompletion.cpp

```diff
@@ -2080,6 +2078,23 @@ class CompletionLookup final : public swift::VisibleDeclConsumer {
     llvm_unreachable("unhandled kind");
   }
 
+  bool isUnresolvedMemberIdealType(Type Ty) {
+    assert(Ty);
+    if (!IsUnresolvedMember)
+      return false;
+    Type idealTy = expectedTypeContext.idealType;
+    if (!idealTy)
+      return false;
+    /// Consider optional object type is the ideal.
+    /// For exmaple:
+    ///   enum MyEnum { case foo, bar }
+    ///   func foo(_: MyEnum?)
+    ///   fooo(.<HERE>)
+    /// Prefer '.foo' and '.bar' over '.some' and '.none'.
+    idealTy = idealTy->lookThroughAllOptionalTypes();
+    return idealTy->isEqual(Ty);
```

Comment:
Why `isEqual` instead of `isConvertible`?

Reply:
Returning convertible types by static member is pretty uncommon, I guess.
```swift
class Derived: Base {}
extension Base {
  static func createSub() -> Derived
}
```
I just didn't want to check them all by relatively heavy `isConvertible()` just for prioritization. But if you think it's worth it, I'm happy to do it.

Reply:
It's fine, just making sure I understand.